commands/run.go

```diff
@@ -0,0 +1,207 @@
+		faas-cli run stronghash -f ./stronghash.yml
+		faas-cli run stronghash -f ./stronghash.yml --port 8081
+		`,
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+
+			opts.output = cmd.OutOrStdout()
+			opts.err = cmd.ErrOrStderr()
+
+			return runFunction(ctx, args[0], opts)
+		},
+		// TODO: unhide once we are happy with the DX.
+		Hidden: true,
+	}
+
+	cmd.Flags().BoolVar(&opts.print, "print", false, "Print the docker command instead of running it")
+	cmd.Flags().IntVarP(&opts.port, "port", "p", 8080, "port to bind the function to")
+	cmd.Flags().StringVar(&opts.network, "network", "", "connect function to an existing network, use 'host' to access other process already running on localhost. When using this, '--port' is ignored, if you have port collisions, you may change the port using '-e port=NEW_PORT'")
+	cmd.Flags().StringToStringVarP(&opts.extraEnv, "env", "e", map[string]string{}, "additional environment variables (ENVVAR=VALUE), use this to experiment with different values for your function")
+
+	return cmd
+}
+
+func runFunction(ctx context.Context, name string, opts runOptions) error {
+	services, err := stack.ParseYAMLFile(yamlFile, "", name, true)
+	if err != nil {
+		return err
+	}
+
+	if len(services.Functions) > 1 {
+		return fmt.Errorf("multiple functions matching %q in the stack file", name)
+	}
+
+	fnc := services.Functions[name]
+	// TODO: we should probably use a levelled logger here
+	// fmt.Fprintf(opts.output, "%#v\n\n", fnc)
+
+	cmd, err := buildDockerRun(ctx, fnc, opts)
+	if err != nil {
+		return err
+	}
+
+	if opts.print {
+		fmt.Fprintf(opts.output, "%s\n", cmd.String())
+		return nil
+	}
+
+	cmd.Stdout = opts.output
+	cmd.Stderr = opts.err
+
+	err = cmd.Start()
+	if err != nil {
+		return err
+	}
+
+	return cmd.Wait()
+}
+
+// buildDockerRun constructs a exec.Cmd from the given stack Function
+func buildDockerRun(ctx context.Context, fnc stack.Function, opts runOptions) (*exec.Cmd, error) {
+	args := []string{"run", "--rm", "-i", fmt.Sprintf("-p=%d:8080", opts.port)}
+
+	if opts.network != "" {
+		args = append(args, fmt.Sprintf("--network=%s", opts.network))
+	}
+
+	fprocess, err := deriveFprocess(fnc)
+	if err != nil {
+		return nil, err
+	}
+
+	for name, value := range fnc.Environment {
+		args = append(args, fmt.Sprintf("-e=%s=%s", name, value))
+	}
+
+	moreEnv, err := readFiles(fnc.EnvironmentFile)
+	if err != nil {
+		return nil, err
+	}
+
+	for name, value := range moreEnv {
+		args = append(args, fmt.Sprintf("-e=%s=%s", name, value))
+	}
+
+	for name, value := range opts.extraEnv {
+		args = append(args, fmt.Sprintf("-e=%s=%s", name, value))
+	}
+
+	if fnc.ReadOnlyRootFilesystem {
+		args = append(args, "--read-only")
+	}
+
+	if fnc.Limits != nil {
+		if fnc.Limits.Memory != "" {
+			// use a soft limit for debugging
+			args = append(args, fmt.Sprintf("--memory-reservation=%s", fnc.Limits.Memory))
+		}
+
+		if fnc.Limits.CPU != "" {
```

Comment:
`fnc.Limits.CPU` is usually written as `100Mi` for instance.

Does that work here, with --cpus? What does docker expect?